openmetadata-ui/src/main/resources/ui/src/components/common/DeleteWidget/DeleteWidgetModal.tsx

```diff
@@ -114,12 +119,14 @@ const DeleteWidgetModal = ({
     try {
       setIsLoading(false);
       setEntityDeleteState((prev) => ({ ...prev, loading: 'waiting' }));
-      const response = await deleteEntity(
-        prepareType ? prepareEntityType(entityType) : entityType,
-        entityId ?? '',
-        Boolean(isRecursiveDelete),
-        !entityDeleteState.softDelete
-      );
+      const response = await (api
+        ? api()
+        : deleteEntity(
+            prepareType ? prepareEntityType(entityType) : entityType,
```

Comment:
There might be a code smell here because of the nested ternary operators. Can you please check and take care of that? Please use the SonarLint extension for catching the smells while coding.